File: backend/TrainMateX.Domain/Entities/Exercise.cs
public class Exercise
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public Category Category { get; set; }
    public MuscleGroup MuscleGroup { get; set; }
    public Equipment Equipment { get; set; }
    public Difficulty Difficulty { get; set; }

    public ICollection<WorkoutExercise> WorkoutExercises { get; set; } = new List<WorkoutExercise>();
}

File: frontend/src/types/Exercise.ts
export interface Exercise {
  id: string
  name: string
  description: string
  muscleGroup: MuscleGroup
  equipment: Equipment
  difficulty: Difficulty
  videoUrl?: string
  thumbnailUrl?: string
}

export type MuscleGroup =
  | 'CHEST'
  | 'BACK'
  | 'SHOULDERS'
  | 'LEGS'
  | 'ARMS'
  | 'CORE'
  | 'FULL_BODY'
  | 'CARDIO'

export type Equipment =
  | 'BARBELL'
  | 'DUMBBELL'
  | 'MACHINE'
  | 'CABLE'
  | 'BODYWEIGHT'
  | 'KETTLEBELL'
  | 'RESISTANCE_BAND'
  | 'OTHER'

export type Difficulty = 'BEGINNER' | 'INTERMEDIATE' | 'ADVANCED'
